tests/sqlite_provenance.py

```
@@ -0,0 +1,144 @@
+            Proxy for sqlite3.connect that records provenance metadata for each created connection.
+
+            Records metadata (connection id, database path, current test nodeid, thread name/id, and creation stack) in the tracker registry keyed by the connection object's id.
+
+            Parameters:
+                *args: Positional arguments forwarded to sqlite3.connect (first positional arg is the database path).
+                **kwargs: Keyword arguments forwarded to sqlite3.connect (may include "database").
+
+            Returns:
+                sqlite3.Connection: The connection object returned by the underlying sqlite3.connect call.
+            """
+            caller_factory = kwargs.get("factory", None)
```

Ruff (0.15.12)

[warning] Use `kwargs.get("factory")` instead of `kwargs.get("factory", None)`

Replace `kwargs.get("factory", None)` with `kwargs.get("factory")`

(SIM910)